app/Services/NotificationService.php

```
@@ -0,0 +1,290 @@
+    private function replaceShortcodes($template, array $shortcodes): array
+    {
+        $replacements = array_merge($this->defaultShortcodes(), $shortcodes);  // combine shortcodes
+
+        return [
+            'subject' => $this->replaceText($template['subject'] ?? '', $replacements),
+            'content' => $this->replaceText($template['content'] ?? '', $replacements),
+            'name' => $shortcodes['name'] ?? '',
+            'title' => $this->replaceText($template['title'] ?? '', $replacements),
+            'message' => $this->replaceText($template['message'] ?? '', $replacements),
+        ];
+    }
```

PHPMD (2.15.0)

Avoid unused private methods such as 'replaceShortcodes'. (Unused Code Rules)

(UnusedPrivateMethod)